conans/client/toolchain/cmake.py

```diff
@@ -0,0 +1,184 @@
+        {%- endfor %}
+        
+        # Set some environment variables
+        {%- for it, value in environment.items() %}
+        set(ENV{{ '{' }}{{ it }}{{ '}' }} "{{ value }}")
+        {%- endfor %}
+        
+        include("{{ conan_adjustements_cmake }}")
+        
+        get_property( _CMAKE_IN_TRY_COMPILE GLOBAL PROPERTY IN_TRY_COMPILE )
+        if(NOT _CMAKE_IN_TRY_COMPILE)
+            message(">>>> NOT TRY COMPILE")
+            set(CMAKE_PROJECT_INCLUDE "{{ conan_project_include_cmake }}")  # Will be executed after the project
+            include("{{ conan_adjustements_cmake }}")
+
+            # We are going to adjust automagically many things as requested by Conan
+            #   these are the things done by 'conan_basic_setup()'
+            
+            conan_set_flags("")
+            conan_set_flags("_RELEASE")
+            conan_set_flags("_DEBUG")
+            # TODO: Do we want to set flags here if we are going to use targets? Maybe these flags are defined inside the 'toolchain' method of the recipe
+            
+            conan_set_rpath()
+            conan_set_std()
+            conan_set_fpic()
+            
+            #conan_check_compiler()
+            conan_set_libcxx()
+            #conan_set_vs_runtime()  # This goes to the after-project() command call
+            conan_set_find_paths()
+            conan_set_find_library_paths()
+            
+            set(CMAKE_CXX_FLAGS_INIT "${CONAN_CXX_FLAGS}" CACHE STRING "" FORCE)
+            set(CMAKE_C_FLAGS_INIT "${CONAN_C_FLAGS}" CACHE STRING "" FORCE)
+            set(CMAKE_SHARED_LINKER_FLAGS_INIT "${CONAN_SHARED_LINKER_FLAGS}" CACHE STRING "" FORCE)
+            set(CMAKE_EXE_LINKER_FLAGS_INIT "${CONAN_EXE_LINKER_FLAGS}" CACHE STRING "" FORCE)
+        else()
+            # message(">>>> TRY COMPILE")
+        endif()
+        
+        conan_set_compiler()
+
+
+        {#
+        # Host machine        
+        set(CMAKE_SYSTEM_NAME {{host.os}})
+        set(CMAKE_SYSTEM_VERSION {{host.os_version}})
+        set(CMAKE_SYSTEM_PROCESSOR {{host.arch}})
+
+        # Build machine (only if different)
+        set(CMAKE_HOST_SYSTEM_NAME {{build.os}})
+        set(CMAKE_HOST_SYSTEM_VERSION {{build.os_version}})
+        set(CMAKE_HOST_SYSTEM_PROCESSOR {{build.arch}})
+        
+        set(CMAKE_C_COMPILER {{c_compiler}})
+        set(CMAKE_CXX_COMPILER {{cxx_compiler}})
+        #}
+    """)
+
+    def __init__(self, conanfile,
+                 generator=None,
+                 cmake_system_name=True,
+                 parallel=True,
+                 build_type=None,
+                 toolset=None,
+                 make_program=None,
+                 msbuild_verbosity="minimal",
+                 # cmake_program=None,  # TODO: cmake program should be considered in the environment
+                 generator_platform=None
+                 ):
+        self._conanfile = conanfile
+        del conanfile
```

Comment:
:thinking: whot?

Reply:
just to be sure that I was not using it below. Let's leave these things for the actual code review (probably I will remove it before)